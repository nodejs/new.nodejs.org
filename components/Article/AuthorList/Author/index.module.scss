.link {
  &:hover img {
    transform: scale(1.1);
  }

  img {
    background-size: contain;
    border: 2px solid var(--brand-light);
    border-radius: 100%;
    display: block;
    height: 30px;
    margin-top: 5px;
    transition: all 0.2s ease-in-out;
    width: 30px;
  }
}
