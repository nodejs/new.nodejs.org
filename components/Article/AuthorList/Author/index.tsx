import { useState } from 'react';
import { useIntl } from 'react-intl';
import Image from 'next/image';
import styles from './index.module.scss';
import type { FC } from 'react';

type AuthorProps = { username: string; size?: number };

const Author: FC<AuthorProps> = ({ username, size }) => {
  const githubUserName = username.trim();
  const githubLink = `https://github.com/${githubUserName}`;
  const githubImgLink = `https://github.com/${githubUserName}.png?size=${size}`;

  const intl = useIntl();

  const [authorImg, setAuthorImg] = useState(githubImgLink);

  const translation = intl.formatMessage(
    { id: 'components.article.author.githubLinkLabel' },
    { username }
  );

  return (
    <a
      className={styles.link}
      href={githubLink}
      aria-label={translation}
      key={username}
      target="_blank"
      rel="noopener noreferrer"
    >
      <Image
        alt={githubUserName}
        src={authorImg}
        placeholder="blur"
        blurDataURL="/placeholder-img.png"
        width={size}
        height={size}
        onError={() => setAuthorImg('/placeholder-img.png')}
      />
    </a>
  );
};

export default Author;
