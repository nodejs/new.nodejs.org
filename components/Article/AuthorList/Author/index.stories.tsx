import Author from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof Author>;
type Meta = MetaObj<typeof Author>;

export const Default: Story = {
  args: {
    username: 'nodejs',
    size: 60,
  },
};

export const WithourUsername: Story = {
  args: {
    username: '',
    size: 0,
  },
};

export default { component: Author } as Meta;
