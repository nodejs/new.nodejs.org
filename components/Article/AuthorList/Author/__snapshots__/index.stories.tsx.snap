// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article/AuthorList/Author Default smoke-test 1`] = `
<a class="Author_link__6oI83"
   href="https://github.com/nodejs"
   aria-label="nodejs Github - opens in new tab"
   target="_blank"
   rel="noopener noreferrer"
>
  <img alt="nodejs"
       loading="lazy"
       width="60"
       height="60"
       decoding="async"
       data-nimg="1"
       srcset="https://github.com/nodejs.png?size=60?w=64&amp;q=75 1x, https://github.com/nodejs.png?size=60?w=128&amp;q=75 2x"
       src="https://github.com/nodejs.png?size=60?w=128&amp;q=75"
       style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat; background-image: url(&quot;/placeholder-img.png&quot;);"
  >
</a>
`;

exports[`Article/AuthorList/Author WithourUsername smoke-test 1`] = `
<a class="Author_link__6oI83"
   href="https://github.com/"
   aria-label=" Github - opens in new tab"
   target="_blank"
   rel="noopener noreferrer"
>
  <img alt
       loading="lazy"
       width="0"
       height="0"
       decoding="async"
       data-nimg="1"
       srcset="https://github.com/.png?size=0?w=16&amp;q=75 1x"
       src="https://github.com/.png?size=0?w=16&amp;q=75"
       style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat; background-image: url(&quot;/placeholder-img.png&quot;);"
  >
</a>
`;
